test/integration/load_balancers/custom_lb_policy.h

```diff
@@ -24,7 +24,12 @@ class ThreadAwareLbImpl : public Upstream::ThreadAwareLoadBalancer {
   public:
     LbImpl(const Upstream::HostSharedPtr& host) : host_(host) {}
 
-    Upstream::HostConstSharedPtr chooseHost(Upstream::LoadBalancerContext*) override {
+    Upstream::HostConstSharedPtr chooseHost(Upstream::LoadBalancerContext* context) override {
+      if (context && context->requestStreamInfo()) {
```

Comment:
should dynamic metadata be populated in this method here or rather at a call site? Does it need to be populated by default for all callers/in all tests that pass non null context?

Reply:
I see how it can be confusing. I have created a dedicated test LB that writes into metadata in the `test/common/upstream/cluster_manager_impl_test.cc` test suite and reverted changes to the one in the test/integration